logic/src/commonMain/kotlin/com/wire/kalium/logic/data/asset/Asset.kt

```diff
@@ -1,5 +1,21 @@
 package com.wire.kalium.logic.data.asset
 
-abstract class Asset
+data class UploadAssetId(val key: String)
 
-data class PublicAsset(val key: String) : Asset()
+/**
+ * On creation of this model, the use case should "calculate" the logic.
+ * For example, rules that apply for an eternal/public asset
+ */
+data class UploadAssetMetadata(
+    val mimeType: String,
+    val isPublic: Boolean,
+    val retentionType: RetentionType
+)
```

Comment:
is there any issue by having data and metadata of an asset at the same place

```suggestion
data class UploadAssetData(
    val data: ByteArray,
    val mimeType: AssetType, // where AssetType is an enum type of supported assets
    val isPublic: Boolean,
    val retentionType: Int
) {
     var md5: String
        private set
    init {
        md5 = calcMd5(data)
    }
    
    // auto generated code by the IDE to calculate the equals and hashCode 
    // because of the ByteArray

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || this::class != other::class) return false

        other as UploadAssetMetadata

        if (!data.contentEquals(other.data)) return false
        if (mimeType != other.mimeType) return false
        if (isPublic != other.isPublic) return false
        if (retentionType != other.retentionType) return false
        if (md5 != other.md5) return false

        return true
    }

    override fun hashCode(): Int {
        var result = data.contentHashCode()
        result = 31 * result + mimeType.hashCode()
        result = 31 * result + isPublic.hashCode()
        result = 31 * result + retentionType
        result = 31 * result + md5.hashCode()
        return result
    }
}

```

Reply:
Great suggestion, I updated with a proposal and a sealed class for mimetypes.